### PR TITLE
ref(jira): Clean up and differentiate errors raised by Jira webhooks

### PR DESCRIPTION
**Pull request by lobsterkatie**
This updates the way we deal with errors which arise while handling incoming webhook requests. Key changes:
- Stop logging "unclear jira exception" exceptions for errors we've seen before. This both prevents duplicates in sentry (one copy of the error with the log statement, one without) and lets us identify new ones.
- Stop capturing Atlassian automated pen-testing `MethodNotAllowed` errors as sentry events and instead just log them.
- In cases where we have HTML or XML as our error message, move that to a context  and replace the message with a more human-friendly one. This requires adding a `__str__` method to `ApiError`, so that the message change will be picked up by the Sentry SDK. (`Exception` doesn't have a `text` attribute, and so its `__str__` method of course ignores the `text` attribute on `ApiError`.)

Ref: WOR-2926

### CODE REVIEW

File: src/sentry/integrations/jira/webhooks/base.py
Comment (armenzg) [inline src/sentry/integrations/jira/webhooks/base.py:57]: Yes, this is the way 🎉

File: src/sentry/integrations/jira/webhooks/base.py
Comment (armenzg) [inline src/sentry/integrations/jira/webhooks/base.py:67]: This one as a tag is useful. Maybe `jira.host`?

File: src/sentry/integrations/jira/webhooks/base.py
Comment (armenzg) [inline src/sentry/integrations/jira/webhooks/base.py:91]: Very creative on modifying the `exc.text` 👍🏻

File: src/sentry/integrations/jira/webhooks/base.py
Comment (armenzg) [inline src/sentry/integrations/jira/webhooks/base.py:95]: Maybe just a warning message? I think it may be useful during local dev.

**Review by armenzg — APPROVED** [review]
Very thorough work specially on the writing tests.
Thanks @lobsterkatie !

File: src/sentry/integrations/jira/webhooks/base.py
Comment (lobsterkatie) [reply src/sentry/integrations/jira/webhooks/base.py:95]: All of these which run through `super` get logged, because that's part of what it does. I've updated the comment just below this to reflect that.

File: src/sentry/integrations/jira/webhooks/base.py
Comment (lobsterkatie) [reply src/sentry/integrations/jira/webhooks/base.py]: Done in https://github.com/getsentry/sentry/commit/a9d45fc05e6c5d255c1d6eb6ce759a3b1997e31e. It requires a rebase onto https://github.com/getsentry/sentry/pull/47777, so I'll hold off merging this until that's through. (That's why there are suddenly a ton of extra commits here. They'll go away once that's merged.)

**Review by armenzg — APPROVED** [review]

**Pull request merged**
